posts/2019/07/uarray-update-api-changes-overhead-and-comparison-to-__array_function__.ipynb

```diff
@@ -0,0 +1,580 @@
+    "    @staticmethod\n",
+    "    def __ua_convert__(dispatchables, coerce):\n",
+    "        converted = []\n",
+    "        for d in dispatchables:\n",
+    "            if d.type is numbers.Number and isinstance(d.value, CustomNumber):\n",
+    "                # Access the internal value, no conversion going on\n",
+    "                converted.append(d.value.n)\n",
+    "            else:\n",
+    "                # Handle the base case\n",
+    "                return NotImplemented\n",
+    "        \n",
+    "        return converted"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "('Custom', 'int', '<lambda>', (1,), {'b': 2})\n"
+     ]
+    }
+   ],
+   "source": [
+    "with ua.set_backend(DispatchBackendCustom), ua.set_backend(DispatchBackendInt):\n",
+    "    print(my_multimethod_with_dispatch(CustomNumber(1), 2))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Permanent registration of backend\n",
+    "`uarray` provides permanent registration for backends, after which the backend will be automatically tried every time. This is user-facing code, and we recommend that no libraries register themselves permanently, apart from reference implementations. This can be done via the [`register_backend`](https://uarray.readthedocs.io/en/latest/generated/uarray.register_backend.html) method."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Comparison with `__array_function__`\n",
+    "### Replacement and dispatching based on different objects\n",
+    "In this section, let’s define a function with `__array_function__`, its `uarray` equivalent and explore the limitations of `__array_function__` and how `uarray` attempts to resolve them. Here's a snippet from NumPy showing roughly how `np.sum` would be defined ([current code as of writing here](https://github.com/numpy/numpy/blob/02c8e80d5f65c7870f71c989c425d1bad24bd312/numpy/core/fromnumeric.py#L2040-L2182)):\n",
+    "\n",
+    "```python\n",
+    "def _sum_dispatcher(a, axis=None, dtype=None, out=None, keepdims=None,\n",
+    "                    initial=None, where=None):\n",
+    "    return (a, out)\n",
+    "\n",
+    "\n",
+    "@array_function_dispatch(_sum_dispatcher)\n",
+    "def sum(a, axis=None, dtype=None, out=None, keepdims=np._NoValue,\n",
+    "        initial=np._NoValue, where=np._NoValue):\n",
+    "    # Default implementation\n",
+    "```\n",
+    "\n",
+    "Here's the equivalent `uarray` code:\n",
+    "\n",
+    "```python\n",
+    "def _sum_ar(a, kw, d):\n",
+    "    def replacer(a, axis=None, dtype=None, out=None,\n",
+    "        keepdims=None, initial=None, where=None):\n",
+    "        return (d[0],), {\n",
+    "            'out': d[1],\n",
+    "            'dtype': d[2],\n",
+    "            'axis': d[3],\n",
```

Comment:
nit: this would read better if reformatted:
- indent second line of signature so `keepdims` and `a` line up
- fit the return dict on a single line (and are the `\n`'s needed?)

Reply:
Done.